src/lib/pricingEngine.js

```diff
@@ -0,0 +1,64 @@
+const { BigNumber } = require('ethers');
+
+const { calculateProductDataForTranche } = require('./helpers');
+const { selectProductPools, selectProduct } = require('../store/selectors');
+
+/**
+ * Calculates the pricing information for a given product based on its associated pools,
+ * weighted by each pool's available capacity.
+ *
+ * @param {Object} store - The application state store containing product data.
+ * @param {number} productId - The unique identifier for the product to calculate pricing for.
+ * @returns {Object|null} An object containing the product ID, an array of price per pool,
+ *                       and the weighted average price, or null if no pools are found.
+ */
+function pricingEngine(store, productId) {
+  const { assets, assetRates } = store.getState();
+  const product = selectProduct(store, productId);
+  const productPools = selectProductPools(store, productId);
+  const now = BigNumber.from(Math.floor(Date.now() / 1000));
+
+  if (!product || !productPools.length) {
+    return null;
+  }
+
+  const pricePerPool = [];
+  let totalWeight = BigNumber.from(0);
+  let weightedSum = BigNumber.from(0);
+
+  // Get capacity data per pool
+  const { capacityPerPool } = calculateProductDataForTranche(
+    productPools,
+    0, // use trancheIndex 0 to include all active tranches in pool capacity calculation
+    product.useFixedPrice,
+    now,
+    assets,
+    assetRates,
+  );
+
+  productPools.forEach((pool, index) => {
+    pricePerPool.push({
```

Comment:
@rackstar should we have a check for `targetWeight !== 0` here? I'm thinking the pool pricing when weight is 0 is not useful 🤔  or at least add the `targetWeight` value too so we can filter it on FE. wdyt?

Reply:
@mixplore @MilGard91 is it possible for a pool to have `targetWeight: 0` but have `availableCapacity > 0`?

in which case it might be better to only skip on:

```javascript
if (pool.targetWeight.isZero() && availableCapacity.isZero()) {
  continue;
}
```

Reply:
https://github.com/NexusMutual/cover-router/pull/149/commits/d5f9ca625d6112ae37cfb4f56c617e264e5412c2

Reply:
on second thought i think we should just check `availableCapacity` instead of targetWeight. in the case that `targetWeight: 0` and `availableCapacity > 0` it should still be include it in the weighting average.

Reply:
latest change: https://github.com/NexusMutual/cover-router/pull/149/commits/54c5d982c026cebd839f40531fe1bf64e075fdae